### ISSUE TITLE
FURB168 has false positives on empty tuples

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#19047]
### Summary

[`isinstance-type-none` (FURB168)](https://docs.astral.sh/ruff/rules/isinstance-type-none/) has false positives when the tuples it checks are empty. `()` is not the same as `None`.

[Playground](https://play.ruff.rs/c699f540-a4ca-4b66-aca9-e6929587f222)

```console
$ cat >furb168.py <<'# EOF'
from typing import Union
print(isinstance(None, ()))
try:
    print(isinstance(None, Union[()]))
except TypeError as e:
    print(f"{type(e).__name__}: {e}")
# EOF

$ python furb168.py
False
TypeError: Cannot take a Union of no types.

$ ruff --isolated check furb168.py --select FURB168 --fix
Found 2 errors (2 fixed, 0 remaining).

$ cat furb168.py
from typing import Union
print((None is None))
try:
    print((None is None))
except TypeError as e:
    print(f"{type(e).__name__}: {e}")

$ python furb168.py
True
True
```

### Version

ruff 0.12.1 (32c54189c 2025-06-26)

**Comment by MeGaGiGaGon** [comment]
I think I found the source of the issue:
https://github.com/astral-sh/ruff/blob/34052a1185392963c465737e9647f404abdfe8d5/crates/ruff_linter/src/rules/refurb/rules/isinstance_type_none.rs#L102-L104
The `all` of an empty iterable is `true`, so `is_none` returns `true` on an empty tuple.
